R/time_event.R

```diff
@@ -99,32 +106,39 @@ prep.step_time_event <- function(x, training, info = NULL, ...) {
     trained = TRUE,
     rules = x$rules,
     columns = col_names,
+    keep_original_cols = get_keep_original_cols(x),
     skip = x$skip,
     id = x$id
   )
 }
 
 #' @export
 bake.step_time_event <- function(object, new_data, ...) {
-  if (length(object$column) == 0L) {
-    # Empty selection
-    return(new_data)
-  }
+  col_names <- names(object$columns)
+  check_new_data(col_names, object, new_data)
+
+  for (col_name in col_names){
+    tmp <- get_time_events(
+      rules = object$rules,
+      column = object$columns[col_name],
+      name = col_name,
+      new_data = new_data
+    )
 
-  new_cols <- purrr::imap_dfc(object$columns, time_event_helper,
-                              new_data, object$rules)
+    names(tmp) <- paste(col_name, names(tmp), sep = "_")
+    tmp[] <- lapply(X = tmp, FUN = vctrs::vec_cast, integer())
 
-  new_cols <- check_name(new_cols, new_data, object, names(new_cols))
+    tmp <- check_name(tmp, new_data, object, names(tmp))
+    new_data <- vctrs::vec_cbind(new_data, tmp)
+  }
 
-  new_data <- dplyr::bind_cols(new_data, new_cols)
-  new_data <- dplyr::select(new_data, -names(object$columns))
+  new_data <- remove_original_cols(new_data, object, col_names)
   new_data
 }
 
-time_event_helper <- function(columnn, name, new_data, rule) {
-  res <- purrr::map_dfc(rule, ~ alma_in(new_data[[columnn]], .x))
-
-  names(res) <- paste(name, names(res), sep = "_")
+get_time_events <- function(rules, column, name, new_data) {
+  res <- lapply(X = rules, FUN = \(x) almanac::alma_in(new_data[[column]],x))
```

Comment:
We can't use the `\(x)` shorthand yet since it would lock out past versions of R. 

```suggestion
  res <- lapply(X = rules, FUN = function(x) almanac::alma_in(new_data[[column]], x))
```